### PR TITLE
🔨 [#312][d] - Give stable identity to Context Provider values 🧠

### LINKED ISSUE #312: 🔨 [Maintainability] React Context Provider values should have stable identities (typescript:S6481)
## Summary

SonarCloud flagged **3** occurrence(s) of this rule in `sushigo-webapp`.

- **Rule:** `typescript:S6481`
- **Category:** Maintainability
- **Severity:** MAJOR

## Affected locations

- `src/components/ui/toast-provider.tsx:51`
- `src/contexts/SidebarContext.tsx:18`
- `src/contexts/ThemeContext.tsx:26`

## Proposed approach

Wrap the object/array passed as `value` to a Context Provider in `useMemo` (and any functions in `useCallback`) so consumers don't re-render on every parent render due to a new value identity.

## References

- SonarCloud project: [pakodiazdev_sushigo-webapp](https://sonarcloud.io/project/issues?id=pakodiazdev_sushigo-webapp&rules=typescript:S6481)

### 📊 Estimates
- **Optimistic:** `30m`
- **Pessimistic:** `1.5h`
- **Tracked:** `0.27h`

### 📅 Sessions
```json
[
  { "date": "2026-07-28", "start": "19:34", "end": "19:50" }
]
```

## 📊 Retrospective
- **Actual total:** 16m (16m)
- **vs optimistic:** −14m
- **vs pessimistic:** −1h14m

**Justification:** The fix was mechanical and scoped exactly as anticipated — wrap each Provider's `value` in `useMemo` and its inline handlers in `useCallback`, no design decisions or unexpected edge cases involved. Two of the three test files already had partial coverage (`SidebarContext.test.tsx`, `ThemeContext.test.tsx`), so only one small stability assertion needed adding per file rather than building suites from scratch.

### PR DESCRIPTION
**Pull request by pakodiazdev**
## Summary
Closes #312
Devin Review: https://deepwiki.com/pakodiazdev/sushigo/pull/348

- 🧠 Wrap `SidebarContext`, `ThemeContext`, and `ToastContext` Provider `value` in `useMemo`
- 🧠 Wrap the inline toggle handlers in `SidebarContext`/`ThemeContext` in `useCallback` (toast-provider's handlers were already `useCallback`-wrapped)
- 🧪 Add value-identity stability tests to all three providers' test suites

Resolves SonarCloud rule `typescript:S6481` (3 occurrences) — Provider values previously created a new object identity on every render, forcing consumers to re-render unnecessarily.

## Test plan
- [x] Vitest: `npx vitest run src/contexts/__tests__/SidebarContext.test.tsx src/contexts/__tests__/ThemeContext.test.tsx src/components/ui/__tests__/toast-provider.test.tsx` — 33/33 passing
- [x] Full Vitest suite: `npx vitest run` — 242 files, 3546 passed, 3 pre-existing skipped
- [x] Linters: ESLint ✅ (0 errors, 2 pre-existing unrelated warnings) · TypeScript ✅

## Manual Testing
No user-facing behavior change — this is a pure internal refactor for render-performance/maintainability.

1. Run `npx vitest run` in `code/webapp` and confirm all tests pass (see Test plan above)
2. Optionally, in the browser dev tools React Profiler: open the app, toggle the sidebar/theme/a toast, and confirm consumers of `SidebarContext`/`ThemeContext`/`ToastContext` no longer re-render on unrelated parent re-renders

## Workspace
`sushigo-d` — `refactor/312-stable-context-values`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: doc/tasks/2026-07/312-stable-context-values.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  { "date": "2026-07-28", "start": "19:34", "end": "19:50" }
+]
+```
+
+---
+
+## 🔗 References
+
+- GitHub issue: [#312](https://github.com/pakodiazdev/sushigo/issues/312)
+- SonarCloud: https://sonarcloud.io/project/issues?id=pakodiazdev_sushigo-webapp&rules=typescript:S6481
```
[the remark this answers is not in the document]
Comment (pakodiazdev) [reply doc/tasks/2026-07/312-stable-context-values.md:75]: Added the mandatory `## 📊 Retrospective` section comparing tracked time (16m) against the optimistic/pessimistic estimates, with a short justification. Fixed in fe56de7 — add retrospective section to task #312's file.
